cmd/gen/makefile/flag.go

```diff
@@ -16,21 +16,29 @@ const (
 )
 
 type flag struct {
-	Flavour  string
-	Language string
+	Flavours gen.FlavourSlice
+	Language gen.Language
 }
 
 func (f *flag) Init(cmd *cobra.Command) {
-	cmd.Flags().StringVarP(&f.Flavour, flagFlavour, "f", "", fmt.Sprintf(`The type of project that you want to generate the Makefile for. Possible values: <%s>`, strings.Join(gen.AllFlavours(), "|")))
-	cmd.Flags().StringVarP(&f.Language, flagLanguage, "l", "", fmt.Sprintf(`The programming language of project that you want to generate the Makefile for. Possible values: <%s>`, strings.Join(gen.AllLanguages(), "|")))
+	cmd.Flags().VarP(gen.NewFlavourSliceFlagValue(&f.Flavours, []gen.Flavour{}), flagFlavour, "f", fmt.Sprintf(`List of types of project that you want to generate the Makefile for. Possible values: <%s>`, strings.Join(gen.AllFlavours(), "|")))
+	cmd.Flags().VarP(gen.NewLanguageFlagValue(&f.Language, gen.Language("")), flagLanguage, "l", fmt.Sprintf(`The programming language of project that you want to generate the Makefile for. Possible values: <%s>`, strings.Join(gen.AllLanguages(), "|")))
 }
 
 func (f *flag) Validate() error {
-	if !gen.IsValidFlavour(f.Flavour) {
-		return microerror.Maskf(invalidFlagError, "--%s must be one of <%s>", flagFlavour, strings.Join(gen.AllFlavours(), "|"))
+	if len(f.Flavours) == 0 {
+		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagFlavour)
```

Comment:
Would there be a way to still show allowed values? E.g. `must be one or more of ...`

Reply:
Feel free to suggest something. `gen.AllFlavours()` is still there. I thought it's enough to have them printed in `--help`.

Reply:
I find the best errors are those that tell me how to fix them 😉 

```suggestion
		return microerror.Maskf(invalidFlagError, "--%s must be one or more of: %s", flagFlavour, strings.Join(gen.AllFlavours(), ", ")
```